src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -63,6 +63,104 @@ public static void BuildChain()
             }
         }
 
+#if netcoreapp11
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [Fact]
+        public static void VerifyChainFromHandle()
+        {
+            X509Chain chain;
+
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var microsoftDotComIssuer = new X509Certificate2(TestData.MicrosoftDotComIssuerBytes))
+            using (var microsoftDotComRoot = new X509Certificate2(TestData.MicrosoftDotComRootBytes))
+            using (var unrelated = new X509Certificate2(TestData.DssCer))
+            using (var chainHolder = new ChainHolder())
+            {
+                chain = chainHolder.Chain;
+
+                chain.ChainPolicy.ExtraStore.Add(unrelated);
+                chain.ChainPolicy.ExtraStore.Add(microsoftDotComRoot);
+                chain.ChainPolicy.ExtraStore.Add(microsoftDotComIssuer);
+                chain.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+                chain.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+
+                bool valid = chain.Build(microsoftDotCom);
+                Assert.True(valid, "Chain built validly");
+
+                X509Chain chain2 = new X509Chain(chain.ChainContext);
+                Assert.Equal(chain2.ChainContext, chain.ChainContext);
+
+                Assert.Equal(3, chain2.ChainElements.Count);
+                Assert.Equal(microsoftDotCom, chain2.ChainElements[0].Certificate);
+                Assert.Equal(microsoftDotComIssuer, chain2.ChainElements[1].Certificate);
+                Assert.Equal(microsoftDotComRoot, chain2.ChainElements[2].Certificate);
+
+                chain2.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
+                chain2.ChainPolicy.VerificationTime = new DateTime(2015, 10, 15, 12, 01, 01, DateTimeKind.Local);
+                chain2.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                valid = chain2.Build(microsoftDotCom);
+                Assert.True(valid, "Chain built validly");
```

Comment:
Please use different messages here and on line 89.  For example "Original chain built validly" and "Cloned chain built validly"